ddtrace/span.py

```diff
@@ -169,6 +169,14 @@ def start(self, value):
         # type: (Union[int, float]) -> None
         self.start_ns = int(value * 1e9)
 
+    @property
+    def resource(self):
+        return self._resource[0]
+
+    @resource.setter
+    def resource(self, value):
+        self._resource[0] = value
+
```

Comment:
This will likely be quite expensive at runtime, but we probably don't have a way around this if we can't afford retaining traces in memory ☹️

Reply:
From the artifacts attached to the [django-overhead-profile](https://github.com/DataDog/dd-trace-py/actions/runs/1241334870) job

![Screenshot 2021-09-17 at 14 37 10](https://user-images.githubusercontent.com/20231758/133791680-adb28f44-e600-4708-8f07-5d7738a1b1ec.png)

This is quite a small sample, but it still shows up in profiles (and probably the simple Django app that runs in CI doesn't stress the new code paths a lot).

Reply:
I agree this is not great for performance overhead, but from all the existing options we have right now, this is probably our best.

We are ok moving forward with this approach for now, but we will want to discuss a better pattern to have this shared data between tracing and profiling in the future.